pkg/federate/create_or_update_federator.go

```diff
@@ -65,20 +72,75 @@ func (f *createOrUpdateFederator) Distribute(ctx context.Context, obj runtime.Ob
 
 	f.prepareResourceForSync(toDistribute)
 
-	result, err := util.CreateOrUpdate[*unstructured.Unstructured](ctx, resource.ForDynamic(resourceClient), toDistribute,
-		func(obj *unstructured.Unstructured) (*unstructured.Unstructured, error) {
-			return util.CopyImmutableMetadata(obj, toDistribute), nil
+	result, newObj, err := util.CreateOrUpdateWithOptions[*unstructured.Unstructured](ctx,
+		util.CreateOrUpdateOptions[*unstructured.Unstructured]{
+			Client:            resource.ForDynamic(resourceClient),
+			Obj:               toDistribute,
+			IdentifyingLabels: f.getIdentifyingLabels(toDistribute),
+			MutateOnUpdate: func(obj *unstructured.Unstructured) (*unstructured.Unstructured, error) {
+				return util.CopyImmutableMetadata(obj, toDistribute), nil
+			},
 		})
 
 	if f.eventLogName != "" {
 		if result == util.OperationResultCreated {
 			logger.Infof("%s: Created %s \"%s/%s\" ", f.eventLogName, toDistribute.GetKind(), toDistribute.GetNamespace(),
-				toDistribute.GetName())
+				newObj.GetName())
 		} else if result == util.OperationResultUpdated {
 			logger.Infof("%s: Updated %s \"%s/%s\" ", f.eventLogName, toDistribute.GetKind(), toDistribute.GetNamespace(),
-				toDistribute.GetName())
+				newObj.GetName())
 		}
```

Comment:
This won’t make a practical difference, but for consistency, it would be better to use `newObj` throughout instead of mixing `toDistribute` and `newObj` (since the created or updated object is `newObj`).